File: 1. Basic Topics/1. Arrays/cpp/1572. Matrix Diagonal Sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int result = 0;
        int length = mat.size();
        for(int i = 0; i<length; i++){
            int j = i;

            result += mat[i][j];

            j = length-i-1;

            if(j!=i){
                result += mat[i][j];
            }
        }

        return result;
    }
};

File: 1. Basic Topics/1. Arrays/py/1572. Matrix Diagonal Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        result = 0

        for row in range(len(mat)):
            col = row

            result += mat[row][col]

            col = len(mat)-row-1

            if(col!=row):
                result += mat[row][col]

        return result